Spack installation fails due to issue with mpark-variant

**Describe the bug**
When trying to install `openpmd-api` with spack, installation fails. The installation log reports some error related to the version (1.4.0 vs 1.3.0) of `mpark_variant`.

**To Reproduce**
```
spack install openpmd-api
```
(More, precisely, in my case, `spack spec openpmd-api` returns:
```
openpmd-api@develop%gcc@7.2.0~adios1~adios2 build_type=RelWithDebInfo +hdf5+json+mpi~python arch=linux-ubuntu16.04-x86_64 
    ^cmake@3.13.4%gcc@7.2.0~doc+ncurses+openssl~ownlibs~qt arch=linux-ubuntu16.04-x86_64 
        ^bzip2@1.0.6%gcc@7.2.0+shared arch=linux-ubuntu16.04-x86_64 
            ^diffutils@3.7%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
        ^curl@7.63.0%gcc@7.2.0~darwinssl~libssh~libssh2~nghttp2 arch=linux-ubuntu16.04-x86_64 
            ^openssl@1.1.1b%gcc@7.2.0+systemcerts arch=linux-ubuntu16.04-x86_64 
                ^perl@5.26.2%gcc@7.2.0+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads arch=linux-ubuntu16.04-x86_64 
                    ^gdbm@1.18.1%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                        ^readline@7.0%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                            ^ncurses@6.1%gcc@7.2.0~symlinks~termlib arch=linux-ubuntu16.04-x86_64 
                                ^pkgconf@1.5.4%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                ^zlib@1.2.11%gcc@7.2.0+optimize+pic+shared arch=linux-ubuntu16.04-x86_64 
        ^expat@2.2.5%gcc@7.2.0+libbsd arch=linux-ubuntu16.04-x86_64 
            ^libbsd@0.9.1%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
        ^libarchive@3.3.2%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
            ^libxml2@2.9.8%gcc@7.2.0~python arch=linux-ubuntu16.04-x86_64 
                ^libiconv@1.15%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                ^xz@5.2.4%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
            ^lz4@1.8.1.2%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
            ^lzo@2.09%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
            ^nettle@3.4%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                ^gmp@6.1.2%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                    ^autoconf@2.69%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                        ^m4@1.4.18%gcc@7.2.0 patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00,c0a408fbffb7255fcc75e26bd8edab116fc81d216bfd18b473668b7739a4158e,fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c8 +sigsegv arch=linux-ubuntu16.04-x86_64 
                            ^libsigsegv@2.11%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                    ^automake@1.16.1%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                    ^libtool@2.4.6%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
        ^libuv@1.25.0%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
        ^rhash@1.3.5%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
    ^hdf5@1.10.4%gcc@7.2.0~cxx~debug~fortran~hl+mpi+pic+shared~szip~threadsafe arch=linux-ubuntu16.04-x86_64 
        ^openmpi@3.1.3%gcc@7.2.0~cuda+cxx_exceptions fabrics=auto ~java~legacylaunchers~memchecker~pmi schedulers=auto ~sqlite3~thread_multiple+vt arch=linux-ubuntu16.04-x86_64 
            ^hwloc@1.11.11%gcc@7.2.0~cairo~cuda~gl+libxml2~nvml+pci+shared arch=linux-ubuntu16.04-x86_64 
                ^libpciaccess@0.13.5%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                    ^util-macros@1.19.1%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
                ^numactl@2.0.12%gcc@7.2.0 arch=linux-ubuntu16.04-x86_64 
    ^mpark-variant@1.4.0%gcc@7.2.0 build_type=RelWithDebInfo arch=linux-ubuntu16.04-x86_64 
    ^nlohmann-json@3.5.0%gcc@7.2.0 build_type=RelWithDebInfo +single_header+test arch=linux-ubuntu16.04-x86_64 
```

In this case, the installation fails with:
```
  >> 24    CMake Error at CMakeLists.txt:375 (find_package):
     25      Could not find a configuration file for package "mpark_variant" that is
     26      compatible with requested version "1.4.0".
     27    
     28      The following configuration files were considered but not accepted:
     29    
     30        /home/rlehe/spack/opt/spack/linux-ubuntu16.04-x86_64/gcc-7.2.0/m
           park-variant-1.4.0-b3yztpyr4s4wrrnephthv4adtsw27qql/lib/cmake/mpark_
           variant/mpark_variant-config.cmake, version: 1.3.0
```
This is quite confusing, since `spack info` reports that I have `mpark-variant@1.4.0` installed. 

My understanding is that `mpark-variant` is shipped along with `openpmd-api`. Any idea what might be happening here? Also, has `spack install` been tested since #465

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -372,7 +372,7 @@ if(openPMD_USE_INTERNAL_VARIANT)
     )
     message(STATUS "MPark.Variant: Using INTERNAL version 1.4.0")
 else()
-    find_package(mpark_variant 1.4.0 REQUIRED)
+    find_package(mpark_variant 1.3.0 REQUIRED) # TODO: we want 1.4.1+
     target_link_libraries(openPMD::thirdparty::mpark_variant
         INTERFACE mpark_variant)
     message(STATUS "MPark.Variant: Found version ${mpark_variant_VERSION}")

